src/js/player.js

```diff
@@ -2209,54 +2209,59 @@ class Player extends Component {
    *        The callback that should be called when the techs play is actually called
    */
   play_(callback = silencePromise) {
-    // If this is called while we have a play queued up on a loadstart, remove
-    // that listener to avoid getting in a potentially bad state.
-    if (this.playOnLoadstart_) {
-      this.off('loadstart', this.playOnLoadstart_);
-    }
-
-    // If the player/tech is not ready, queue up another call to `play()` for
-    // when it is. This will loop back into this method for another attempt at
-    // playback when the tech is ready.
-    if (!this.isReady_) {
+    this.playSuccessCallbacks_.push(callback);
```

Comment:
I think the name here can be confusing. While we don't handle rejecting the promise ourselves yet (we really should add it at some point), it doesn't mean that play was successful, just that it wasn't terminated, since we call it with the promise from `play()` which may end up rejecting the promise.

Reply:
renamed this variable and the function for it `playCallbacks_`